src/app/pages/common/ix-forms/services/ix-formatter.service.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { createServiceFactory, SpectatorService } from '@ngneat/spectator';
+import { IxFormatterService } from 'app/pages/common/ix-forms/services/ix-formatter.service';
+
+describe('IxFormatterService', () => {
+  let spectator: SpectatorService<IxFormatterService>;
+
+  const createService = createServiceFactory({
+    service: IxFormatterService,
+  });
+
+  beforeEach(() => {
+    spectator = createService();
+  });
+
+  it('should instantiate', () => {
+    expect(spectator).toBeTruthy();
+  });
+
+  describe('memorySizeFormatting', () => {
+    it('should return empty string when invalid size string is passed', () => {
+      const formatted = spectator.service.memorySizeFormatting('2u');
+      expect(formatted).toBe('');
+    });
+
+    it('should return normalized string when valid size string is passed', () => {
+      const formatted = spectator.service.memorySizeFormatting('2gb');
+      expect(formatted).toBe('2 GiB');
+    });
+
+    it('should return the value with B when no unit is provided', () => {
+      const formatted = spectator.service.memorySizeFormatting('2');
+      expect(formatted).toBe('2 B');
+    });
+  });
+
+  describe('memorySizeParsing', () => {
+    it('should return NaN when invalid size string is passed', () => {
+      const parsed = spectator.service.memorySizeParsing('2u');
+      expect(parsed).toBe(NaN);
```

Comment:
I don't think we should use NaN. No one will remember to check for NaN and `typeof NaN` is still `number`. Let's use `null` or `undefined`.